Sources/FoundationEssentials/CMakeLists.txt

```diff
@@ -50,8 +50,26 @@ add_subdirectory(TimeZone)
 add_subdirectory(URL)
 
 if(SwiftFoundation_MACRO)
-  target_compile_options(FoundationEssentials PRIVATE
-    "SHELL:-plugin-path ${SwiftFoundation_MACRO}")
+    message(STATUS "SwiftFoundation_MACRO provided, using macros in ${SwiftFoundation_MACRO}")
+    # A path to Foundation macros was provided, so we use that path
+    target_compile_options(FoundationEssentials PRIVATE
+        "SHELL:-plugin-path ${SwiftFoundation_MACRO}")
+else()
+    message(STATUS "SwiftFoundation_MACRO not provided, building Foundation macros locally for host")
+    # No path to Foundation macros was provided, so we must build it ourselves
+    set(FoundationMacros_BuildLocalExecutable YES)
+    FetchContent_Declare(FoundationMacros
+        SOURCE_DIR ${CMAKE_SOURCE_DIR}/Sources/FoundationMacros)
```

Comment:
Wouldn't `CMAKE_SOURCE_DIR` be the `swift-corelibs-foundation` when this project is added as an external project from there? Would `${CMAKE_CURRENT_SOURCE_DIR}/../FoundationMacros` be more correct here?